controllers/grid/settings/sections/form/PKPSectionForm.php

```diff
@@ -138,22 +139,32 @@ public function fetch($request, $template = null, $display = false)
 
     public function initData()
     {
+        $subeditorUserGroups = [];
+
         if ($this->getSection() !== null) {
-            $this->setData([
-                'assignedSubeditors' => Repo::user()
-                    ->getCollector()
-                    ->filterByContextIds([Application::get()->getRequest()->getContext()->getId()])
-                    ->filterByRoleIds($this->assignableRoles)
-                    ->assignedToSectionIds([$this->getSectionId()])
-                    ->getIds()
-                    ->toArray(),
-            ]);
-        } else {
-            $this->setData([
-                'assignedSubeditors' => [],
-            ]);
+            $contextId = Application::get()->getRequest()->getContext()->getId();
+            $assignedSubeditors = Repo::user()
+                ->getCollector()
+                ->filterByContextIds([$contextId])
+                ->filterByRoleIds($this->assignableRoles)
+                ->assignedToSectionIds([$this->getSectionId()])
+                ->getIds()
+                ->toArray();
+
+            if (!empty($assignedSubeditors)) {
+                $subEditorsDao = DAORegistry::getDAO('SubEditorsDAO'); /** @var SubEditorsDAO $subEditorsDao */
+                $subeditorUserGroups = $subEditorsDao->getAssignedUserGroupIds(
```

Comment:
Rather than passing a list of user IDs in here based on the previous query, why not just add the role filter to this function and do it all in one place?

Reply:
I also thought about that but later decided not to do that as that makes the DAO method too big . Moved it .

Reply:
Reverted the changes as same issue spotted for `Category` . So the `DAO::getAssignedUserGroupIds()` can not have a generalised implementation as we need to filter by section and category, 2 different entity .